ios/RNLinksdk.m

```diff
@@ -93,107 +88,85 @@ - (void)stopObserving {
-        if (isUsingLinkToken) {
-            self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithLinkToken:linkTokenInput
-                                                                               oauthStateId:oauthStateId
-                                                                              configuration:linkConfiguration
-                                                                                   delegate:self.linkViewDelegate];
-        }
-        else if ([linkTokenInput hasPrefix:kRNLinkKitItemAddTokenPrefix]) {
-            self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithItemAddToken:linkTokenInput
-                                                                                 configuration:linkConfiguration
-                                                                                      delegate:self.linkViewDelegate];
-        }
-        else {
-            self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithPublicToken:linkTokenInput
-                                                                                configuration:linkConfiguration
-                                                                                     delegate:self.linkViewDelegate];
-        }
-    }
-    else if ([institution length] > 0) {
-        self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithInstitution:institution
-                                                                            configuration:linkConfiguration
-                                                                                 delegate:self.linkViewDelegate];
-    }
-    else if ([paymentTokenInput length] > 0) {
-        self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithPaymentToken:paymentTokenInput
-                                                                              oauthStateId:oauthStateId
-                                                                             configuration:linkConfiguration
-                                                                                  delegate:self.linkViewDelegate];
-    }
-    else if ([oauthStateId length] > 0) {
-        self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithOAuthStateId:oauthStateId
-                                                                             configuration:linkConfiguration
-                                                                                  delegate:self.linkViewDelegate];
+    if (isUsingLinkToken) {
+        PLKLinkTokenConfiguration *config = [self getLinkTokenConfiguration:configuration
+                                                           onSuccessHandler:onSuccess];
+        config.onEvent = onEvent;
+        config.onExit = onExit;
+        self.linkHandler = [PLKPlaid createWithLinkTokenConfiguration:config
+                                                                error:NULL];
+    } else {
```

Comment:
Will this handle the case where public key and token are null?

Reply:
No – what is the expected behaviour if public key is null? Should we exit immediately with an error?

Reply:
Verified behaviour: if public key is nil then we'll get a requires_credential exit status